pkg/utils/otel_test.go

```
@@ -519,3 +476,79 @@ func TestSetupOTelSDKWithConfig_IPEndpoint(t *testing.T) {
+	s := newSampler(cfg) // default = parentbased_traceidratio
+
+	// Without parent, ratio=0.0 should drop.
+	result := s.ShouldSample(trace.SamplingParameters{ParentContext: context.Background()})
+	if result.Decision != trace.Drop {
+		t.Errorf("expected Drop without parent at ratio 0.0, got %v", result.Decision)
+	}
+
+	// With a sampled parent span (created via tracer), child should also be sampled
+	// even though ratio=0.0, because parent-based sampler honors the parent's decision.
+	tp := trace.NewTracerProvider(trace.WithSampler(trace.AlwaysSample()))
+	defer tp.ForceFlush(context.Background())
```

golangci-lint (2.12.2)

[error] Error return value of `tp.ForceFlush` is not checked

(errcheck)